src/main/java/org/wso2/carbon/inbound/googlepubsub/GooglePubSubMessageConsumer.java

```diff
@@ -0,0 +1,331 @@
+
+import com.google.api.gax.core.ExecutorProvider;
+import com.google.api.gax.core.FixedCredentialsProvider;
+import com.google.api.gax.core.InstantiatingExecutorProvider;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.pubsub.v1.*;
+import org.apache.axiom.om.OMElement;
+import org.apache.axis2.builder.Builder;
+import org.apache.axis2.builder.BuilderUtil;
+import org.apache.axis2.builder.SOAPBuilder;
+import org.apache.axis2.transport.TransportUtils;
+import org.apache.commons.io.input.AutoCloseInputStream;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.SynapseException;
+import org.apache.synapse.core.SynapseEnvironment;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.mediators.base.SequenceMediator;
+import org.wso2.carbon.inbound.endpoint.protocol.generic.GenericPollingConsumer;
+
+import com.google.api.gax.batching.FlowControlSettings;
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.pubsub.v1.DeadLetterPolicy;
+import com.google.cloud.pubsub.v1.Subscriber;
+import com.google.common.util.concurrent.MoreExecutors;
+
+import java.io.ByteArrayInputStream;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class GooglePubSubMessageConsumer extends GenericPollingConsumer {
+    private static final Log log = LogFactory.getLog(GooglePubSubMessageConsumer.class);
+    private String projectId;
+    private String contentType;
+    private String subscriptionId;
+    private GooglePubSubSubscriptionManager subscriptionManager;
+    private Subscriber subscriber = null;
+    private boolean isConsumed = false;
+    private boolean isPolling = false;
+
+    private long maxOutstandingMessageCount;
+    private long maxOutstandingMessageSize;
+
+    private static final AtomicInteger threadId = new AtomicInteger(1);
+
+    private int concurrentConsumers;
+    private String keyFilePath;
+    private GoogleCredentials credentials;
+    private boolean createSubscriptionOnConnect;
+    private boolean updateSubscriptionIfExists;
+
+    private int executorThreadsPerConsumers;
+    private String endpoint;
+
+    public GooglePubSubMessageConsumer(Properties properties, String name, SynapseEnvironment synapseEnvironment,
+            long scanInterval, String injectingSeq, String onErrorSeq, boolean coordination, boolean sequential) {
+        super(properties, name, synapseEnvironment, scanInterval, injectingSeq, onErrorSeq, coordination, sequential);
```

Comment:
Shall we add a new line at the start of the methods? We can use this wso2 codestyle and format the file in one go. https://raw.githubusercontent.com/wso2/product-is/refs/heads/master/docs/wso2_codestyle.xml

Ref: https://somindagamage.medium.com/how-to-configure-checkstyle-and-findbugs-plugins-to-intellij-idea-38148aad2387